app/src/main/java/net/gsantner/markor/activity/DocumentEditFragment.java

```diff
@@ -671,6 +702,83 @@ public boolean saveDocument(final boolean forceSaveEmpty) {
         }
     }
 
+    /**
+     * This method saves the current document as a new document.
+     * 1. The name of the new document is specified by the user by a dialog.
+     * 2. The user will then edit the new document in a new Document edit fragment.
+     * 3. When the user gets back to from the new document, he/she can continue to edit the old document in this fragment.
+     */
+    public boolean saveAsNewDocument() {
+        File currentFolder = new File(this._document.getPath()).getParentFile();
+        Log.d(getFragmentTag(), "saveAsNewDocument: find currentFolder is " + currentFolder);
+        if (currentFolder == null) {
+            return false;
+        }
+        NewFileDialog dialog =
+            NewFileDialog.newInstance(currentFolder.getAbsoluteFile(), false, (ok, newFile) -> {
```

Comment:
Can we just use SearchOrCustomTextDialog here? Will simplify things IMO

Reply:
So I need to add a new method in the class SearchOrCustomTextDialogCreator to create a dialog for user to specify the name, right?